### DISCUSSION TITLE
Option to disable upload in method save

### DISCUSSION THREAD

**Discussion by ThomasBouche** [datapane/datapane#196] [Ideas]
Hi, 
I'm working with my team on open source Python libraries (we publish Shapash (https://github.com/MAIF/shapash/tree/master/shapash) and Melusine (https://github.com/MAIF/melusine)) and we are going to open source a new library. In this new library, the main output is an HTML report. 
To generate this HTML report, we would like to use your library which we find interesting and easy to use.
However, when we save the report with the save method, Datapane returns an upload message.
Would it be possible to have an option to disable this upload like :  .save(upload = False) ?

**Comment by lanthias** [comment]
Hi @ThomasBouche, apologies for the late response here.

I think you're absolutely right and we will be removing this, and similar messages encouraging save users to use upload, in an upcoming release. I'll keep you in the loop.

**Reply by ThomasBouche** [reply]
Hello, great,
In the meantime, we have used the _save method in our library. We will be able to take the new version into account during the release.

For your information, if all goes well, we will release the open source tomorrow. You can see our work there.
we are looking forward to further discuss with you the functionnalities of datapane to generate a report.
